src/subbuffers.cpp

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2021 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/subbuffers.hpp>
+#include <miopen/miopen.h>
+
+#if MIOPEN_BACKEND_OPENCL
+#include <miopen/handle.hpp>
+#include <miopen/ocldeviceinfo.hpp>
+#endif
+
+#include <tuple>
+
+namespace miopen {
+
+std::size_t GetSubbufferAlignment(const miopen::Handle* handle)
```

Comment:
Could it be possible to use ```struct MultiBufferWorkspaceTraits``` instead?
ref: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/54ac76851bfaedf8321f8b8ec0a3c9c74bb48f33/src/include/miopen/buffer_info.hpp#L311

Reply:
The `GetSubbufferAlignment` returns the hardware constant of subbuffer alignment if you pass a handle and a constant expected value otherwise. The reason is the following: OpenCL has a meaningful value there, meaning that you have to align a subbuffer when it is being created unless you want a runtime error. Handle is optional here due to the pooling API flaw I have mentioned in the PR descriptions.

Reply:
OK, got you.